src/main/java/seedu/duke/lesson/LessonList.java

```diff
@@ -35,4 +38,24 @@ public void clearLessonList() {
     public void deleteLesson(int index) {
         lessonList.remove(index);
     }
+
+    public Lesson getLesson(int index) throws DukeException {
+        try {
+            return lessonList.get(index);
+        } catch (IndexOutOfBoundsException exception) {
+            throw new DukeException(Message.ERROR_INDEX_OUT_OF_BOUNDS);
+        } catch (NumberFormatException exception) {
+            throw new DukeException(Message.ERROR_NUMBER_FORMAT);
+        }
+    }
+
+    public LessonList findLesson(String keyword) {

```

Comment:
`findLessonsByKeyword` or `filterLessonsByKeyword` might be a better naming?

Reply:
Similar to `findTasks`, the Streams implemenation of this would be more elegant. But currently it's ok, we will move on to implement more important stuffs first.